validation-test/stdlib/Collection/LazyFilterCollection.swift.gyb

```diff
@@ -0,0 +1,88 @@
+// -*- swift -*-
+
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %S/../../../utils/gyb %s -o %t/main.swift
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-build-swift %t/main.swift -o %t/LazyFilterCollection.swift.a.out
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-run %t/LazyFilterCollection.swift.a.out
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+
+// Also import modules which are used by StdlibUnittest internally. This
+// workaround is needed to link all required libraries in case we compile
+// StdlibUnittest with -sil-serialize-all.
+import SwiftPrivate
+#if _runtime(_ObjC)
+import ObjectiveC
+#endif
+
+var CollectionTests = TestSuite("Collection")
+
+%{
+variations = [('', 'Sequence'), ('', 'Collection'), ('Bidirectional', 'Collection')]
+}%
+
+// Test collections using value types as elements.
+% for (specifier, kind) in variations:
+CollectionTests.add${specifier}${kind}Tests(
+  make${kind}: { (elements: [OpaqueValue<Int>]) in
+    Minimal${specifier}${kind}(elements: elements).lazy.filter { _ in return true }
```

Comment:
Thank you for the update!

Given the diagnosis in https://bugs.swift.org/browse/SR-1278, we can't really tell which overload is being called here (after all, Array is a bidirectional collection).  To make sure, could you add an explicit return type to closures in these tests?